google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/CopyBackupRequest.java

```diff
@@ -0,0 +1,143 @@
+
+import com.google.api.core.InternalApi;
+import com.google.cloud.bigtable.admin.v2.internal.NameUtil;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.protobuf.util.Timestamps;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.threeten.bp.Instant;
+
+/** Fluent wrapper for {@link com.google.bigtable.admin.v2.CopyBackupRequest} */
+public final class CopyBackupRequest {
+  private final com.google.bigtable.admin.v2.CopyBackupRequest.Builder requestBuilder =
+      com.google.bigtable.admin.v2.CopyBackupRequest.newBuilder();
+  private final String sourceBackupId;
+  private final String sourceClusterId;
+  private final String sourceInstanceId;
+  private final String sourceProjectId;
+
+  private String destClusterId;
+
+  /**
+   * Create a {@link CopyBackupRequest} object. It assumes the source backup is located in the same
+   * instance as the destination backup. To copy from a backup in another instance, use {@link
+   * #of(String, String, String) of} method. To copy from a backup in another project, use {@link
+   * #of(String, String, String, String) of} method
+   */
+  public static CopyBackupRequest of(String clusterId, String backupId) {
+    CopyBackupRequest request = new CopyBackupRequest(clusterId, backupId, null, null);
+    return request;
+  }
+
+  /**
+   * Create a {@link CopyBackupRequest} object. It assumes the source backup is located in the same
+   * project as the destination backup. To copy from a backup in another project, use {@link
+   * #of(String, String, String, String) of} method
+   */
+  public static CopyBackupRequest of(String clusterId, String backupId, String instanceId) {
+    CopyBackupRequest request = new CopyBackupRequest(clusterId, backupId, instanceId, null);
+    return request;
+  }
+
+  /**
+   * Create a {@link CopyBackupRequest} object. The source backup could be located in the same or
+   * different cluster and/or instance and/or project.
+   */
+  public static CopyBackupRequest of(
+      String clusterId, String backupId, String instanceId,  String projectId) {
+    CopyBackupRequest request = new CopyBackupRequest(clusterId, backupId, instanceId, projectId);
+    return request;
+  }
+
+  private CopyBackupRequest(
+      @Nonnull String sourceClusterId,
+      @Nonnull String sourceBackupId,
+      @Nullable String sourceInstanceId,
+      @Nullable String sourceProjectId) {
+    Preconditions.checkNotNull(sourceBackupId);
+    Preconditions.checkNotNull(sourceClusterId);
+    this.sourceClusterId = sourceClusterId;
+    this.sourceBackupId = sourceBackupId;
+    this.sourceInstanceId = sourceInstanceId;
+    this.sourceProjectId = sourceProjectId;
+  }
+
+  public CopyBackupRequest setBackupId(String backupId) {
+    Preconditions.checkNotNull(backupId);
+    requestBuilder.setBackupId(backupId);
+    return this;
+  }
+
+  public CopyBackupRequest setClusterId(String clusterId) {
```

Comment:
maybe rename this to `setDestinationClusterId` and same with the parameter to match the variable and to be clearer to the user?